rule.py

```diff
@@ -150,14 +154,59 @@ def is_interfered_horse(self, old_pos, new_pos):
         elif buffer_x == -2:
             ki_da.x = old_pos.x + buffer_x + 1    
         elif abs(buffer_x) == 1: 
-            ki_da.x = old_pos.x + buffer_x 
+            ki_da.x = old_pos.x 
         else: 
             return False 
         
         return self.board_map_2d[ki_da.y][ki_da.x] == self._IS_EMPTY 
```

Comment:
Here, it is returned true when it is not interfered and return True when it is interfered. Is this what you want? Check the method is_interfered() and make sure it does what you want.

Reply:
Check all the other is_interfered_*() method as well to make sure everything follow the same rule